tasks/main.yml

```diff
@@ -98,30 +80,34 @@
     state: present
     login_user: "{{ database_root_user }}"
     login_password: "{{ database_root_password }}"
+    check_implicit_admin: true
+    login_unix_socket: /run/mysqld/mysqld.sock
 
-## configuration steps to mimic what 'mysql_secure_installation' does:
+## some configuration steps to mimic what 'mysql_secure_installation' does:
 
 - name: Delete anonymous user
   community.mysql.mysql_user:
     user: ""
-    host: "{{ item }}"
+    host_all: yes
     state: absent
     login_user: "{{ database_root_user }}"
     login_password: "{{ database_root_password }}"
-  loop:
-    - "localhost"
-    - "{{ ansible_fqdn }}"
-  no_log: true
+    check_implicit_admin: true
+    login_unix_socket: /run/mysqld/mysqld.sock
 
 - name: Ensure root user can only connect from the local machine
+  community.mysql.mysql_query:
```

Comment:
Does this matter at all if the next step is to force the `root` user to use Unix sockets?

Reply:
Yes, because even after you have forced the `root` user to use the unix_socket plugin you'd also want to clean up entries that still refer to remote hosts because after a (accidental) deactivation of the plugin they'd become active/relevant again